File: 2022.01.24/python/15651.py
n,m = map(int,input().split())

def solution(n,cnt,target,acc):
  if cnt == target:
    return acc+"\n"
  result = ""
  for i in range(1,n+1):
    result += solution(n,cnt+1,target,acc+str(i)+" ")
  return result
print(solution(n,0,m,""))

File: Boj.단계별/14_백트래킹/javascript/15651.js
const [n,m] = require('fs').readFileSync('/dev/stdin').toString().split('\n')[0].split(' ').map(Number);

const solution = (n,cnt,target,acc) => {
  if (cnt == target){
    return acc+"\n";
  }
  let result = "";
  for (let i = 1 ; i<=n ; i++){
    result += solution(n,cnt+1,target,acc+i+" ");
  }
  return result;
}
console.log(solution(n,0,m,""));